examples/distributed_training_with_flexible_runner.py

```diff
@@ -43,8 +42,8 @@ def compute_metrics(self, results):
 def parse_args():
     parser = argparse.ArgumentParser(description='Distributed Training')
     parser.add_argument('--local_rank', '--local-rank', type=int, default=0)
-    parser.add_argument('--use-deepspeed', action='store_true')
-
+    parser.add_argument(
+        '--strategy', choices=['deepspeed', 'fsdp', 'ddp'], default='ddp')
```

Comment:
This script should also support single device training.

Reply:
Set defaults to 'none', which is compatible for single device training and ddp training.